core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSinkProvider.java

```diff
@@ -21,15 +21,35 @@
 {
     ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorOutputTableHandle outputTableHandle);
 
+    default ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorOutputTableHandle outputTableHandle, ConnectorPageSinkId pageSinkId)
```

Comment:
I would suggest marking methods which does not take `ConnectorPageSinkId` as deprecated and migrate all the connectors to new SPI. We would like to remove old overrides from SPI eventually. Can be a separate followup commit